cmd/clusterctl/internal/test/fake_objects.go

```diff
@@ -1092,6 +1093,38 @@ func (f *FakeClusterResourceSet) Objs() []runtime.Object {
 	return objs
 }
 
+type FakeCustomObject struct {
```

Comment:
```suggestion
type FakeExtrernalObject struct {
```
I think that external better represents the goal of this PR (external=not included in the objects managed by clusterctl)